inc/databaseinstance.class.php

```diff
@@ -0,0 +1,221 @@
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+class DatabaseInstance extends CommonDBChild {
+
+   // From CommonDBTM
+   public $auto_message_on_action   = false;
+   static $rightname                   = 'database';
+
+   // From CommonDBChild
+   static public $itemtype       = 'Database';
+   static public $items_id       = 'databases_id';
+
+   static function getTypeName($nb = 0) {
+      return _n('Database instance', 'Database instances', $nb);
+   }
+
+   function defineTabs($options = []) {
+      $ong = [];
+      $this->addDefaultFormTab($ong)
+         ->addImpactTab($ong, $options)
+         ->addStandardTab('Document_Item', $ong, $options)
+         ->addStandardTab('KnowbaseItem_Item', $ong, $options)
+         ->addStandardTab('Notepad', $ong, $options)
+         ->addStandardTab('Log', $ong, $options);
+      return $ong;
+   }
+
+   public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0) {
+      if (!$withtemplate
+          && ($item->getType() == Database::class)
+          && $item->canView()) {
+         $nb = 0;
+         if ($_SESSION['glpishow_count_on_tabs']) {
+            $nb = countElementsInTable(
+               self::getTable(), [
+                  'databases_id' => $item->getID(),
+                  'is_deleted' => 0
+               ]);
+         }
+         return self::createTabEntry(self::getTypeName(), $nb);
+      }
+      return '';
+   }
+
+
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtemplate = 0) {
+      self::showForDatabase($item);
+   }
+
+   /**
+    * Display instances for database
+    *
+    * @param Database $database Database object
+    *
+    * @return void|boolean
+   **/
+   static function showForDatabase(Database $database) {
+
+      $ID = $database->fields['id'];
+
+      if (!$database->getFromDB($ID) || !$database->can($ID, READ)) {
+         return false;
+      }
+      $canedit = $database->canEdit($ID);
+
+      if ($canedit) {
+         echo "<div class='center firstbloc'>".
+                "<a class='vsubmit' href='".static::getFormURL()."?databases_id=$ID'>";
+         echo __('Add an instance');
+         echo "</a></div>\n";
+      }
+
+      echo "<div class='center'>";
+
+      $instances = getAllDataFromTable(
+         self::getTable(), [
+            'WHERE'  => [
+               'databases_id' => $ID,
+               'is_deleted'   => 0
```

Comment:
Maybe deleted instances should be displayed, in order to be able to restore them.